upstairs/src/lib.rs

```diff
@@ -458,8 +482,18 @@ async fn proc(
                             );
                             bail!("expected version 1, got {}", version);
                         }
+
                         negotiated = 1;
-                        deadline = deadline_secs(50);
+
+                        // May be some time between promotion to active, so ping until then
+                        needping = true;
+                    }
+                    Some(Message::Imok) => {
+                        println!("{} client {} is waiting for promotion to active.", up.uuid, up_coms.client_id);
+                    }
+                    Some(Message::YouAreNowActive(uuid)) => {
+                        assert_eq!(uuid, up.uuid);
+                        up.set_active();
```

Comment:
This message required to come back from downstairs as part of the negotiated steps, right?
If so, we should continue to use the `negotiated` variable and add this into the steps (or some other
less of a hack than the method I chose) for how the upstairs/downstairs moves through the connection
process. 
Without it, I believe this step could be skipped.

Reply:
We could do both - Upstairs needs to have it's active bit set so that other operations can check it and error if it's not active, and the use of the negotiated var could be used.